builders/interact/ethereum-api/libraries/web3py.md

```diff
@@ -0,0 +1,264 @@
+---
+title: Send Transactions & Deploy Contracts with Web3.py
+description: Follow this tutorial to learn how to use the Ethereum Web3 Python Library to send transactions and deploy Solidity smart contracts to your Tanssi EVM ContainerChain.
+---
+
+# Web3.py Python Library
+
+## Introduction {: #introduction }
+
+[Web3.py](https://web3py.readthedocs.io/) is a set of libraries that allow developers to interact with Ethereum nodes using HTTP, IPC, or WebSocket protocols with Python. Tanssi EVM ContainerChains have an Ethereum-like API available that is fully compatible with Ethereum-style JSON RPC invocations. Therefore, developers can leverage this compatibility and use the Ethers.js library to interact with a Tanssi EVM ContainerChain node as if they were doing so on Ethereum. For more information on Web3.py, check their [documentation site](https://web3py.readthedocs.io/){target=_blank}.
+
+In this guide, you'll learn how to use setup the Web3.py library for your Tanssi EVM ContainerChain. Next, to showcase the library in action, you'll use Web3.py to send a transaction and deploy a contract on a Tanssi EVM ContainerChain running in Tanssi's [Dancebox](XXX){target=_blank} TestNet. This guide can be adapted for your own Tanssi EVM ContainerChain by simply changing the endpoint.
+
+--8<-- 'text/common/general-py-tutorial-check.md'
+
+
+## Checking Prerequisites {: #checking-prerequisites }
+
+For the examples in this guide, you will need to have the following:
+
+ - An account with funds in the Tanssi EVM ContainerChain you are testing with
+
+## Installing Web3.py {: #install-web3py }
+
+For this guide, you'll need to install the Web3.py library and the Solidity compiler. To install both packages, you can run the following command:
+
+```
```

Comment:
languages should be applied to all code snippets wrapped in triple back ticks. This should be `bash`